src/utils/calendar.ts

```diff
@@ -0,0 +1,34 @@
+import { PersonalizedVenueEvent } from "types/venues";
+import { VenueEvent } from "types/venues";
+import { eventEndTime, eventStartTime } from "utils/event";
+import { WithVenueId } from "utils/id";
+import { getFullVenueInsideUrl } from "utils/url";
+
+import ical from "ical-generator";
+
+export const createCalendar = (
+  allEvents: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[],
+  calendarname: string
+) => {
+  const cal = ical({ name: calendarname });
```

Comment:
Would be good to use named props here as well:

```suggestion
export interface CreateCalendarProps {
  calendarName: string,
  events: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[]
}

export const createCalendar = ({
  calendarName,
  events,
}: CreateCalendarProps) => {
  const cal = ical({ name: calendarName });
```

Reply:
It would also be good to set an explicit return type on this function.

Reply:
Since `PersonalizedVenueEvent` (defined in `src/types/venues.ts`) already extends from `WithVenueId<VenueEvent>`, unless we need the `isSaved` property that it adds (which it doesn't look like we do), we can just simplify the type this function accepts to:

```typescript
export interface CreateCalendarProps {
  calendarName: string,
  events: WithVenueId<VenueEvent>[]
}
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1423/commits/ec68e834fab42e9549a285d7a454e4c29e1f24e4